### PR TITLE
Fixed #460: Test failure on Windows due to CarriageReturn char

### LINKED ISSUE #460: PartialSerializerTest / ChangeSerializer*Test / FileAwareTestLanguageImportTest / IndentationAwareCompletionPrefixProviderTest fail on windows
FileAwareTestLanguageImportTest. testMoveIntoLocalPackage
FileAwareTestLanguageImportTest. testMoveToNewPackage
FileAwareTestLanguageImportTest. testNestedPackage
FileAwareTestLanguageImportTest. testRenameGlobal1
IndentationAwareCompletionPrefixProviderTest. testLastCompleteNode_4_0
IndentationAwareCompletionPrefixProviderTest. testLastCompleteNode_4_0c
IndentationAwareCompletionPrefixProviderTest. testLastCompleteNode_4_0d
IndentationAwareCompletionPrefixProviderTest. testLastCompleteNode_6_0
IndentationAwareCompletionPrefixProviderTest. testLastCompleteNode_6_0b
IndentationAwareCompletionPrefixProviderTest. testLastCompleteNode_7_0
ChangeSerializerTest. testDeleteChild
ChangeSerializerTest. testDeleteTwoChildren
ChangeSerializerTest. testInsertBeforeComment
ChangeSerializerTest. testInsertOneChild
ChangeSerializerTest. testInsertTwoChild
ChangeSerializerTest. testRenameGlobal1
ChangeSerializerTest. testRenameLocal
ChangeSerializerTest. testResourceURIChange
ChangeSerializerTest. testSimple
ChangeSerializerTest. testTwoChildren
ChangeSerializerWithEmfTest. testRefFromXML
ChangeSerializerWithEmfTest. testRefToXML
ChangeSerializerWithImports. testRenameGlobal1
PartialSerializerTest. testMandatoryChildChange
PartialSerializerTest. testMandatoryValueChange
PartialSerializerTest. testMoveInList1
PartialSerializerTest. testOptionalChildChange
PartialSerializerTest. testOptionalChildInsert
PartialSerializerTest. testOptionalChildListInsertIntoEmpty
PartialSerializerTest. testOptionalChildListInsertIntoEmpty2
PartialSerializerTest. testOptionalChildListInsertIntoEndOne
PartialSerializerTest. testOptionalChildListInsertIntoEndThree
PartialSerializerTest. testOptionalChildListInsertIntoEndTwo
PartialSerializerTest. testOptionalChildListInsertIntoFirst
PartialSerializerTest. testOptionalChildListInsertIntoMiddle
PartialSerializerTest. testOptionalChildRemove
PartialSerializerTest. testOptionalChildRemoveListAllOne
PartialSerializerTest. testOptionalChildRemoveListAllTwo
PartialSerializerTest. testOptionalChildRemoveListFirstTwo
PartialSerializerTest. testOptionalChildRemoveListLastTwo
PartialSerializerTest. testOptionalChildRemoveListMiddleTwo
PartialSerializerTest. testOptionalValueChange
PartialSerializerTest. testOptionalValueInsert
PartialSerializerTest. testOptionalValueRemove

### PR DESCRIPTION
**Pull request by meysholdt**

### CODE REVIEW

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/serializer/debug/TextDocumentChangeToString.java [hunk 1]
```diff
@@ -1,96 +1,97 @@
...
-			return "created " + newUri;
-		}
-		if (oldURI.equals(newUri)) {
-			return oldURI.toString();
-		}
-		return "renamed " + oldURI + " to " + newUri;
-	}
-
-}
+/*******************************************************************************
```
Comment (szarnekow) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/serializer/debug/TextDocumentChangeToString.java]: What are the real changes in this file?

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Strings.java [hunk 2]
```diff
@@ -1,608 +1,615 @@
...
-			document.append(word);
-		} else {
-			document.append(line);
-			document.append(ws);
-			document.append(word);
-		}
-		return document.toString();
-	}
-}
+/*******************************************************************************
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/Strings.java]: And here?

File: org.eclipse.xtext/src/org/eclipse/xtext/formatting2/debug/TextRegionsToString.java [hunk 3]
```diff
@@ -1,100 +1,134 @@
...
-		this.text.setTrailingLines(trailingLines);
-		return this;
-	}
-
-	@Override
-	public String toString() {
-		return text + "\n" + list;
-	}
-}
+/*******************************************************************************
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/formatting2/debug/TextRegionsToString.java]: And here?

**Comment by meysholdt** [comment]
> Could you please harmonize this with the clients of org.eclipse.xtend.core.tests.util.LineDelimiters ?
to be done in: https://github.com/eclipse/xtext-xtend/issues/297

**Comment by szarnekow** [comment]
Please make sure that #297 is really done and not another longstanding bug.

**Pull request merged**